prompting/rewards/reward.py

```diff
@@ -51,7 +55,7 @@ def __init__(self, reward_pipeline, agent, response_event, device):
         self.response_event = response_event
         self.device = device
         self.task_rewards = agent.task.reward_definition
-        self.task_penalties = agent.task.penalty_definition
+        self.task_penalties = agent.task.penalty_definition + agent.task.global_penalty_definition
```

Comment:
This causes the penalties to act as a product or a sum?

Reply:
In this case, this results in an array concatenation of penalty definitions (dicts), I leveraged the already existing penalty dynamic to add the global penalties to the same level, where `task_penalties` is composed by the task penalty definitions (Child class) and the task global penalty definitions (Base class).